File: backend/application/infrastructure/persistence/postgres/src/main/java/be/lutske/leolegacy/infrastructure/persistence/repository/CategoryRepository.java
package be.lutske.leolegacy.infrastructure.persistence.repository;

import be.lutske.leolegacy.infrastructure.persistence.entity.CategoryEntity;
import io.quarkus.hibernate.orm.panache.PanacheRepository;
import jakarta.enterprise.context.ApplicationScoped;

import java.util.List;

@ApplicationScoped
public class CategoryRepository implements PanacheRepository<CategoryEntity> {

    public List<CategoryEntity> findAllOrderedByName() {
        return list("ORDER BY name");
    }
}

File: backend/src/main/kotlin/be/lutske/leolegacy/infrastructure/persistence/repository/CategoryRepository.kt
package be.lutske.leolegacy.infrastructure.persistence.repository

import be.lutske.leolegacy.infrastructure.persistence.entity.CategoryEntity
import io.quarkus.hibernate.orm.panache.kotlin.PanacheRepository
import jakarta.enterprise.context.ApplicationScoped

@ApplicationScoped
class CategoryRepository : PanacheRepository<CategoryEntity> {

    fun findAllOrderedByName(): List<CategoryEntity> =
        list("ORDER BY name")
}
